mila_datamodules/cli/prepare_torchvision.py

```diff
@@ -45,56 +58,78 @@ def __call__(
         """
         Path(root).mkdir(parents=True, exist_ok=True)
 
-        constructor_kwargs = constructor_kwargs.copy()  # type: ignore
+        dataset_kwargs = dataset_kwargs.copy()  # type: ignore
         if "download" in inspect.signature(self.dataset_type).parameters:
-            constructor_kwargs["download"] = True
+            dataset_kwargs["download"] = True
 
         logger.debug(
-            f"Using dataset constructor: {self.dataset_type} with args {constructor_args}, and "
-            f"kwargs {constructor_kwargs}"
+            f"Using dataset constructor: {self.dataset_type} with args {dataset_args}, and "
+            f"kwargs {dataset_kwargs}"
         )
-        dataset_instance = self.dataset_type(str(root), *constructor_args, **constructor_kwargs)
+        dataset_instance = self.dataset_type(str(root), *dataset_args, **dataset_kwargs)
         if is_local_main():
             print(dataset_instance)
         return str(root)
 
 
-class SymlinkArchives(PrepareVisionDataset[VD, P]):
-    """Creates symlinks to the archives in the `root` directory."""
+class SymlinkDatasetFiles(AbstractPrepareVisionDataset[VD, P]):
+    """Creates symlinks to the datasets' files in the `root` directory."""
 
     def __init__(
         self,
-        dataset_type: Callable[Concatenate[str, P], VD],
-        relative_paths_to_archives: dict[str, str | Path],
+        source: str | Path = None,
+        files: dict[str, str | Path] = None
     ):
         """
-
         Parameters
         ----------
 
-        - dataset_type:
-            Callable that returns an instance of a `torchvision.datasets.VisionDataset`.
-
-        - relative_paths_to_archives:
-            A mapping from a relative path where the symlink to the archive should be created
+        - source:
+            Source directory to clone the drectories and files hieararchy. If
+            only source is defined, the entire tree will be cloned.
+        - files:
+            A mapping from a path to where the symlink to the archive should be created
             (relative to the 'root' directory) to the actual path to the archive on the cluster.
         """
-        self.dataset_type = dataset_type
-        self.relative_paths_to_archives = {
-            k: Path(v) for k, v in relative_paths_to_archives.items()
-        }
+        if source is not None:
+            source = Path(source).expanduser()
```

Comment:
It might be a good idea to resolve symlinks for example `~/scratch` to `/network/scratch/u/username`
```suggestion
            source = Path(source).expanduser().resolve()
```